google/cloud/bigquery_unified/internal/connection_impl.cc

```diff
@@ -446,10 +447,11 @@ std::shared_ptr<bigquery_unified::Connection> MakeDefaultConnectionImpl(
   // TODO: We should probably add a TracingConnection decorator in order to
   //  associate all the various rpcs that are called as part of these
   //  operations.
-  return std::make_shared<bigquery_unified_internal::ConnectionImpl>(
-      std::move(read_connection), std::move(job_connection),
-      std::move(read_options), std::move(job_options), std::move(job_stub),
-      std::move(background), std::move(options));
+  return MakeTracingConnection(
```

Comment:
This should also be tested, e.g. here in the golden files:

https://github.com/googleapis/google-cloud-cpp/blob/2901ce194f3827c74b463173de369ec17f45eaff/generator/integration_tests/tests/golden_kitchen_sink_connection_test.cc#L374-L420

Calling one RPC and seeing the span is all it takes.

Reply:
Good catch! Added connection_test.cc.